File: Common/InMotionGIT.Common - Copy/DataAcceess/WhereStatements/LogicalOperator.vb
Namespace DataAccess.WhereStatements

    Public Class LogicalOperator(Of t)

        Private ReadOnly tableType As IWhereStatement

        Public Sub New(base As IWhereStatement)
            tableType = base
        End Sub

        Public Function [And]() As t
            tableType.command &= " AND "
            Return tableType
        End Function

        Public Function [Or]() As t
            tableType.command &= " OR "
            Return tableType
        End Function

        Public Function Prepare() As t
            tableType.command = String.Format(" WHERE {0}", tableType.command)

            Return tableType
        End Function

    End Class

End Namespace

File: Common/InMotionGIT.Common/DataAcceess/WhereStatements/LogicalOperator.cs
using Microsoft.VisualBasic.CompilerServices;

namespace InMotionGIT.Common.DataAccess.WhereStatements
{

    public class LogicalOperator<t>
    {

        private readonly IWhereStatement tableType;

        public LogicalOperator(IWhereStatement @base)
        {
            tableType = @base;
        }

        public t And()
        {
            tableType.command += " AND ";
            return Conversions.ToGenericParameter<t>(tableType);
        }

        public t Or()
        {
            tableType.command += " OR ";
            return Conversions.ToGenericParameter<t>(tableType);
        }

        public t Prepare()
        {
            tableType.command = string.Format(" WHERE {0}", tableType.command);

            return Conversions.ToGenericParameter<t>(tableType);
        }

    }

}